lib/src/ctx/context.rs

```diff
@@ -143,8 +143,17 @@ impl<'a> Context<'a> {
 	}
 
 	/// Add a timeout to the context. If the current timeout is sooner than
-	/// the provided timeout, this method does nothing.
+	/// the provided timeout, this method does nothing. If the result of the
+	/// addition causes an overflow, this method does nothing.
+	/// NOTE(gguillemas): This should probably return an option or a result
+	/// to be able to handle overflow cases transparently. This may be done
+	/// for v2.0.0, as it would change the public API.
 	pub fn add_timeout(&mut self, timeout: Duration) {
+		#[cfg(not(target_arch = "wasm32"))]
+		if let Some(deadline) = Instant::now().checked_add(timeout) {
+			self.add_deadline(deadline)
+		}
+		#[cfg(target_arch = "wasm32")]
```

Comment:
@gguillemas `trice@0.4.0` should enable this same functionality in WASM now too, so we can make this code the same.

Reply:
That's great! I will remove the condition and leave the checked addition only.